Objects/Converters/ConverterCSI/ConverterCSIShared/PartialClasses/Geometry/ConvertArea.cs

```diff
@@ -133,46 +139,42 @@ ref tableData
             ref importLog
           );
 
-          if (success == 0)
+          if (success != 0)
+          {
+            appObj.Log.Add(importLog);
+            throw new ConversionException("Failed to apply edited database tables");
+          }
+
+          int numItems = 0;
+          int[] objTypes = null;
+          string[] objNames = null;
+          int[] pointNums = null;
+          foreach (var node in points)
           {
-            int numItems = 0;
-            int[] objTypes = null;
-            string[] objNames = null;
-            int[] pointNums = null;
-            foreach (var node in points)
+            Model.PointObj.GetConnectivity(node, ref numItems, ref objTypes, ref objNames, ref pointNums);
+            if (numItems == 0)
             {
-              Model.PointObj.GetConnectivity(node, ref numItems, ref objTypes, ref objNames, ref pointNums);
-              if (numItems == 0)
-                Model.PointObj.DeleteSpecialPoint(node);
+              Model.PointObj.DeleteSpecialPoint(node);
             }
           }
         }
 #endif
       }
 
       SetAreaProperties(name, area);
-      if (success == 0)
-      {
-        string guid = null;
-        Model.AreaObj.GetGUID(name, ref guid);
-        appObj.Update(
-          status: ApplicationObject.State.Updated,
-          createdId: guid,
-          convertedItem: $"Area{Delimiter}{name}"
-        );
-        if (numErrorMsgs != 0)
-          appObj.Update(
-            log: new List<string>()
-            {
-              $"Area may not have updated successfully. Number of error messages for operation is {numErrorMsgs}",
-              importLog
-            }
-          );
-      }
-      else
+
+      string guid = null;
+      Model.AreaObj.GetGUID(name, ref guid);
```

Comment:
I'm assuming by this point, GetGuid is not going to fail as we've created the area successfully.
Just want to double check this is the case.

If GetGuid could still fail, we should also deal with that successCode

Reply:
It can fail (in that, there is a int return). but I'm unsure the best way to handle it here (and in the may other places we make api calls without checking the result.)

In this case, we've already created/updated the object. The guid is only used to track it for the next update.